src/luagit2/libgit2/lua_libgit2.c

```diff
@@ -0,0 +1,259 @@
+	int size = luaL_checkint(L, 1);
+	check_error_long(git_libgit2_opts(GIT_OPT_SET_MWINDOW_MAPPED_LIMIT, size),
+	    "Error setting memory map limit", NULL);
+	return 1;
+}
+
+
+int lua_GIT_OPT_GET_SEARCH_PATH(lua_State *L) {
+	git_buf path = {0} ;
+	int option_number = luaL_checkint(L, 1);
+
+	switch (option_number) {
+		case 1:
+			check_error_long(git_libgit2_opts(GIT_OPT_GET_SEARCH_PATH, GIT_CONFIG_LEVEL_SYSTEM, &path),
+			    "error getting system level search path", NULL);
+			break;
+		case 2:
+			check_error_long(git_libgit2_opts(GIT_OPT_GET_SEARCH_PATH, GIT_CONFIG_LEVEL_GLOBAL, &path),
+			    "Error getting global level search path", NULL);
+			break;
+		case 3:
+			check_error_long(git_libgit2_opts(GIT_OPT_GET_SEARCH_PATH, GIT_CONFIG_LEVEL_XDG, &path),
+			    "Error getting XDG level search path", NULL);
+			break;
+		case 4:
+			check_error_long(git_libgit2_opts(GIT_OPT_GET_SEARCH_PATH, GIT_CONFIG_LEVEL_PROGRAMDATA, &path),
+			    "Error getting program data level search path", NULL);
+			break;
+	}
+
+	lua_pushstring(L, path.ptr);
+	return 1;
+}
+
+int lua_GIT_OPT_SET_SEARCH_PATH(lua_State *L) {
+	int option_number = luaL_checkint(L, 1);
+	const char *path = luaL_checkstring(L, 2);
+
+	switch (option_number) {
+		case 1:
+			check_error_long(git_libgit2_opts(GIT_OPT_SET_SEARCH_PATH, GIT_CONFIG_LEVEL_SYSTEM, path),
+			    "Error Setting system level search path", NULL);
+			break;
+		case 2:
+			check_error_long(git_libgit2_opts(GIT_OPT_SET_SEARCH_PATH, GIT_CONFIG_LEVEL_GLOBAL, path),
+			    "Error Setting global level search path", NULL);
+			break;
+		case 3:
+			check_error_long(git_libgit2_opts(GIT_OPT_SET_SEARCH_PATH, GIT_CONFIG_LEVEL_XDG, path),
+			    "Error Setting XDG level search path", NULL);
+			break;
+		case 4:
+			check_error_long(git_libgit2_opts(GIT_OPT_SET_SEARCH_PATH, GIT_CONFIG_LEVEL_PROGRAMDATA, path),
+			    "Error Setting program data level search path", NULL);
+			break;
+	}
+	return 1;
+}
+
+
+int lua_GIT_OPT_GET_CACHED_MEMORY(lua_State *L) {
+	ssize_t *current;
+	ssize_t *allowed ;
+	int option_number = luaL_checkint(L, 1);
+
+	check_error_long(git_libgit2_opts(GIT_OPT_GET_CACHED_MEMORY, current, allowed),
+	    "Error getting cached memory size", NULL);
+
+	switch (option_number) {
+		case 1:
+			lua_pushinteger(L, *current);
+			break;
+		case 2:
+			lua_pushinteger(L, *allowed);
+			break;
+	}
+
+	return 1;
+}
+
+int lua_GIT_OPT_SET_CACHE_OBJECT_LIMIT(lua_State *L) {
+	int size = luaL_checkint(L, 1);
+	git_otype type;
+	check_error_long(git_libgit2_opts(GIT_OPT_SET_CACHE_OBJECT_LIMIT, type, size),
+	    "Error setting chache limit", NULL);
+	return 1;
+}
+
+int lua_GIT_OPT_SET_CACHE_MAX_SIZE(lua_State *L) {
+	ssize_t	 size = luaL_checkint(L, 1);
+	check_error_long(git_libgit2_opts(GIT_OPT_SET_CACHE_MAX_SIZE, size),
+	    "Error setting maximum cache size ", NULL);
+	return 1;
+}
+
+int lua_GIT_OPT_ENABLE_CACHING(lua_State *L) {
+	int enabled = luaL_checkint(L, 1);
```

Comment:
While the C libgit library uses an integer for this, maybe in Lua the more idiomatic way would be to pass a boolean?

Reply:
yes. I don't know why I earlier didn't find this `lua_pushboolean`. I do this `lua_pushinteger` -> `lua_pushboolean` on other parts of code too. Thankyou :smile: